inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/KafkaReader.java

```diff
@@ -147,29 +150,14 @@ public Message read() {
 
     @Override
     public boolean isFinished() {
-        if (iterator == null) {
-            // fetch data
-            fetchData(5000);
-            return false;
-        }
-        if (iterator.hasNext()) {
-            lastTime = 0;
-            return false;
-        }
-        // fetch data
-        boolean fetchDataSuccess = fetchData(5000);
-        if (fetchDataSuccess && iterator.hasNext()) {
-            lastTime = 0;
-            return false;
-        } else {
-            if (lastTime == 0) {
-                lastTime = System.currentTimeMillis();
+        if (!isFinished) {
+            if (iterator == null || !iterator.hasNext()) {
+                // fetch data
+                fetchData(5000);
```

Comment:
why fetch data occured in isFinished () ?

Reply:
1.first，fetch data occured in init()
2.After the consumer poll data is read,then continue fetch data, but The premise is that the task does not stop

Reply:
fetch data should be extracted to read() I think